src/Request/FindPointsRequest.php

```diff
@@ -11,20 +11,15 @@ class FindPointsRequest implements Request
 
     private array $searchCriteria;
 
-    public function __construct()
+    public function __construct(array $criteria = [])
     {
-        $this->searchCriteria = [];
-    }
-
-    public function addSearchCriteria(string $type, string $value): void
-    {
-        $this->searchCriteria[$type] = $value;
+        $this->searchCriteria = $criteria;
     }
 
     public function getEndpoint(): string
```

Comment:
consider `getSearchUrl` as method name

Reply:
That name is required by interface

Reply:
Ok, I changed interface to require now getRequestUrl method which I think tells more than getEndpoint